src/Statistics.jl

```diff
@@ -0,0 +1,116 @@
+using Random: shuffle!
+using StatsBase: AbstractWeights, weights, values
+@doc doc"""
+    mean(M,x; weights=1/n*ones(n), stop_iter=100, kwargs... )
+
+computes the Riemannian center of mass also known as Karcher mean of the vector
+`x` of `n` points on the [`Manifold`](@ref) `M`. This function
+uses the gradient descent scheme. Optionally one can provide
+weights $w_i$ for the weighted Riemannian center of mass.
+The general formula to compute the minimizer reads
+````math
+\argmin_{y\in\mathcal M} \frac{1}{2}\sum_{i=1}^n w_i\mathrm{d}_{\mathcal M}^2(y,x_i),
+````
+where $\mathrm{d}_{\mathcal M}$ denotes the Riemannian [`distance`](@ref).
+
+Finally `stop_iter` denotes the maximal number of iterations to perform and
+the `kwargs...` are passed to [`isapprox`](@ref) to stop, when the inimal change
+between two iterates is small. For more stopping criteria
+check the [`Manopt.jl`](https://manoptjl.org) package and use a solver therefrom.
+
+The algorithm is further described in 
+> Afsari, B; Tron, R.; Vidal, R.: On the Convergence of Gradient
+> Descent for Finding the Riemannian Center of Mass,
+> SIAM Journal on Control and Optimization (2013), 51(3), pp. 2230–2260, 
+> doi: [10.1137/12086282X](https://doi.org/10.1137/12086282X),
+> arxiv: [1201.0925](https://arxiv.org/abs/1201.0925">1201.0925)
+"""
+function mean(M::Manifold, x::AbstractVector, w::AbstractWeights = weights(ones(length(x)) / length(x)); kwargs...)
+    y = x[1]
+    return mean!(M, y, x, w; kwargs...)
+end
+mean(M::Euclidean, x, w) = mean(x,w)
+mean(M::Euclidean, x) = mean(x)
+function mean!(M::Manifold, y, x::AbstractVector, w::AbstractWeights = weights(ones(length(x)) / length(x));
+            stop_iter=100,
+            kwargs...
+        ) where {T}
+    yold = y
+    v = zero_tangent_vector.(Ref(M), fill(y,length(x)))
+    for i=1:stop_iter
+        yold = y
+        log!.(Ref(M), v, Ref(yold), x)
+        y = exp(M, yold, sum( values(w).*v ) / 2 )
```

Comment:
```suggestion
        exp!(M, y, yold, sum( values(w).*v ) / 2 )
```
Maybe?
Also, `values(w)` could be calculated before the loop.

Reply:
`values(w)` is just accessing the internal array of values within `AbstractWeights`.

Reply:
OK, then let's keep `values(w)` as is.